aiven/resource_vpc_peering_connection.go

```diff
@@ -286,33 +287,90 @@ func resourceVPCPeeringConnectionRead(_ context.Context, d *schema.ResourceData,
 	return copyVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(d, pc, projectName, vpcID)
 }
 
-func resourceVPCPeeringConnectionDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceVPCPeeringConnectionDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	client := m.(*aiven.Client)
 
 	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+
 	isAzure, err := isAzureVPCPeeringConnection(d, client)
 	if err != nil {
 		return diag.Errorf("Error checking if it Azure VPC peering connection: %s", err)
 	}
 
 	if isAzure {
 		if peerResourceGroup, ok := d.GetOk("peer_resource_group"); ok {
-			err := client.VPCPeeringConnections.DeleteVPCPeeringWithResourceGroup(
-				projectName, vpcID, peerCloudAccount, peerVPC, peerResourceGroup.(string), peerRegion)
-			if err != nil {
+			if err = client.VPCPeeringConnections.DeleteVPCPeeringWithResourceGroup(
+				projectName,
+				vpcID,
+				peerCloudAccount,
+				peerVPC,
+				peerResourceGroup.(string),
+				peerRegion,
+			); err != nil && !aiven.IsNotFound(err) {
 				return diag.Errorf("Error deleting VPC peering connection with resource group: %s", err)
 			}
-			return nil
 		} else {
 			return diag.Errorf("cannot delete an Azure VPC peering connection without `peer_resource_group`")
 		}
 	}
-
-	err = client.VPCPeeringConnections.DeleteVPCPeering(projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
-	if err != nil && !aiven.IsNotFound(err) {
+	if err = client.VPCPeeringConnections.DeleteVPCPeering(
+		projectName,
+		vpcID,
+		peerCloudAccount,
+		peerVPC,
+		peerRegion,
+	); err != nil && !aiven.IsNotFound(err) {
 		return diag.Errorf("Error deleting VPC peering connection: %s", err)
 	}
 
+	if _, err := (&resource.StateChangeConf{
```

Comment:
Just a minor thing: It is hard to read this huge `IF` statement. If we can split it, that would be cool. 

I mean something like:
```go
conf := &resource.StateChangeConf{...}
conf.WaitForStateContext ...
```

Reply:
done